src/Orc.FilterBuilder/Orc.FilterBuilder.Shared/Expressions/StringExpression.cs

```diff
@@ -134,9 +141,9 @@ public override Expression ToLinqExpression(Expression propertyExpr)
                 case Condition.NotIsEmpty:
                     return Expression.NotEqual(propertyExpr, Expression.Constant(string.Empty));
                 case Condition.IsNull:
-                    return Expression.Equal(propertyExpr, Expression.Constant(null));
+                    return Expression.ReferenceEqual(propertyExpr, Expression.Constant(null));
```

Comment:
I've added a test to make sure. This is the way Microsoft suggest using null constants (see https://msdn.microsoft.com/en-us/library/bb349084(v=vs.110).aspx )